Fix thumbnail overflow in render-template

azrikahar opened the pull request:
## Description

Fixes #14926

When we use a large thumbnail in `<render-template>`, it will cause the `<render-template>` wrapper to expand, but still not overflowing the row:

![chrome_H8vJEUdV1u](https://user-images.githubusercontent.com/42867097/185096601-f52e902c-4fdb-4031-b7e5-5be8ca05b1d6.png)

This is because `<render-template`> no longer has `height: 100%` due to PR #14231 to fix text overflow issue.

However when we use image preview display in `<render-display>`, the image's `height: 100%` will properly bound it within the constraint of 32px as it does not has any wrapper:

![chrome_AyOdI2rFGO](https://user-images.githubusercontent.com/42867097/185096680-53a3a9cd-79eb-404e-97df-da11c46ff491.png)

This PR introduces a similar max-height _only_ on `img` elements within `<render-template>` to "revert" the old behavior but still keeping #14231's fix intact:

![chrome_ZU8fekxcba](https://user-images.githubusercontent.com/42867097/185097098-e3f693f6-bb36-41ef-9982-7771d4daeaba.png)


## Type of Change

- [x] Bugfix
- [ ] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated

azrikahar committed:
fix thumbnail overflow in render-template

rijkvanzanten reviewed (changes requested):
No I take that back haha

I don't think we can rely on a hard-coded max height value here. 32px is too big for card-titles for example, and too small for page titles and the "comfortable" table rows 🤔

azrikahar committed:
new fix for render-template thumbnail in table

azrikahar committed:
fix thumbnail not working in page titles

azrikahar committed:
prevent thumbnail overflow in page title

azrikahar commented:
Fiddled with it for a while and I ended up taking a page out of this PR #14644 and moved the css styling to table instead, specifically `<table-row>` in this case. 

I then use the same `v-bind()` that controls the row height, and use it to calculate the height for the `<img>` element, so now it's should the same as display-image and decrease/increase in height based on the row height dynamically. Thoughts on this solution?

||render-template img|display-image|
|---|---|---|
|Compact|![chrome_GazcIndAzT](https://user-images.githubusercontent.com/42867097/185343447-85c54180-1b01-426e-87d6-0ddfa292e3db.png)|![chrome_ACRULlMJim](https://user-images.githubusercontent.com/42867097/185343700-0d37fdbd-efd7-4477-b5af-416944413792.png)|
|Cozy|![chrome_uv57iIIbqO](https://user-images.githubusercontent.com/42867097/185343799-14923b9f-d5e8-440f-924f-423ba77e4416.png)|![chrome_JcxFRHbgGK](https://user-images.githubusercontent.com/42867097/185343836-372e272b-7d6b-4cab-a550-092494947217.png)|
|Comfortable|![chrome_sZ4L9Pg5gO](https://user-images.githubusercontent.com/42867097/185343885-ffbfc460-18d6-4dae-ac1a-6b5bf856c4d5.png)|![chrome_A2SSMWjB3O](https://user-images.githubusercontent.com/42867097/185343917-aaf31543-ae11-4e1a-bda4-9bf7aa464953.png)|

As for thumbnail being used in page title (header bar), I had to update useTemplateData because turns out it doesn't fetch the necessary info for thumbnail fields so we weren't actually able to use thumbnail in page titles before. Updated it to use `adjustFieldsForDisplay` just like how it's being used everywhere else such as relational interfaces:

https://github.com/directus/directus/blob/a803331b4aeaa5c027fe5e847129ad005cbba818/app/src/interfaces/select-dropdown-m2o/select-dropdown-m2o.vue#L159-L162

That being said, I still had to hard code it as 24px (only when used in header bar's page title) or else it will go up to 64px and cover the headline on top. Not sure if there's any other way to derive the value from. Is that okay? 

|enforce 24px|free (up to 64px)|
|---|---|
|![chrome_TGu9ZwI5N0](https://user-images.githubusercontent.com/42867097/185344188-440a09f4-7552-4cd1-b00e-2fe76b2cbafd.png)|![chrome_8QwSmijQkr](https://user-images.githubusercontent.com/42867097/185344263-9d3dd1b8-52be-4991-adf3-83ba6fbd7602.png)|

azrikahar committed:
Merge branch 'main' into issue/14926

azrikahar committed:
Merge branch 'main' into issue/14926
